packages/edit-site/src/components/global-styles/hooks.js

```diff
@@ -0,0 +1,322 @@
+			// If it is not true the content was not escaped and is not safe.
+			if ( ! parsedConfig.isGlobalStylesUserThemeJSON ) {
+				parsedConfig = {};
+			} else {
+				parsedConfig = {
+					...parsedConfig,
+					settings: addUserOriginToSettings( parsedConfig.settings ),
+				};
+			}
+		} catch ( e ) {
+			/* eslint-disable no-console */
+			console.error( 'Global Styles User data is not valid' );
+			console.error( e );
+			/* eslint-enable no-console */
+			parsedConfig = {};
+		}
+
+		return parsedConfig;
+	};
+
+	const config = useMemo( () => {
+		return parseContent( content );
+	}, [ content ] );
+
+	const setConfig = useCallback(
+		( callback ) => {
+			const currentConfig = parseContent(
+				getEditedEntityRecord(
+					'postType',
+					'wp_global_styles',
+					globalStylesId
+				)?.content
+			);
+			const updatedConfig = callback( currentConfig );
+			editEntityRecord( 'postType', 'wp_global_styles', globalStylesId, {
+				content: JSON.stringify( {
+					...updatedConfig,
+					settings: removeUserOriginFromSettings(
+						updatedConfig.settings
+					),
+				} ),
+			} );
+		},
+		[ globalStylesId ]
+	);
+
+	return [ config, setConfig ];
+}
+
+function useGlobalStylesBaseConfig() {
+	const baseConfig = useSelect( ( select ) => {
+		return select( editSiteStore ).getSettings()
+			.__experimentalGlobalStylesBaseStyles;
+	}, [] );
+
+	return baseConfig;
+}
+
+export function useGlobalStylesConfig() {
+	const [ userConfig, setUserConfig ] = useGlobalStylesUserConfig();
+	const baseConfig = useGlobalStylesBaseConfig();
+	const mergedConfig = useMemo( () => {
+		return mergeBaseAndUserConfigs( baseConfig, userConfig );
+	}, [ userConfig, baseConfig ] );
+
+	return [ baseConfig, userConfig, mergedConfig, setUserConfig ];
+}
+
+export const useGlobalStylesReset = () => {
+	const [ config, setConfig ] = useGlobalStylesUserConfig();
+	const canReset = !! config && ! isEqual( config, EMPTY_CONFIG );
+	return [
+		canReset,
+		useCallback( () => setConfig( () => EMPTY_CONFIG ), [ setConfig ] ),
+	];
+};
+
+export function useSetting( path, blockName, source = 'all' ) {
+	const [
+		baseConfig,
+		userConfig,
+		mergedConfig,
+		setUserConfig,
+	] = useGlobalStylesConfig();
+	const finalPath = ! blockName
+		? `settings.${ path }`
+		: `settings.blocks.${ blockName }.${ path }`;
+
+	const getSettingValue = ( configToUse ) => {
+		const result = get( configToUse, finalPath );
+		if ( PATHS_WITH_MERGE[ path ] ) {
+			return result.theme ?? result.core;
+		}
+		return result;
+	};
+
+	const setSetting = ( newValue ) => {
+		setUserConfig( ( currentConfig ) => {
+			const newUserConfig = cloneDeep( currentConfig );
+			set( newUserConfig, finalPath, newValue );
+			return newUserConfig;
+		} );
+	};
+
+	let result;
+	switch ( source ) {
+		case 'all':
+			result = getSettingValue( mergedConfig );
+			break;
+		case 'user':
+			result = getSettingValue( userConfig );
+			break;
+		case 'base':
```

Comment:
Related to another discussion point, I think we can not have only base there are cases where we need to differentiate between 'core', and 'theme', e.g: on the palette we show the default (core), the theme and the user one.

Reply:
Right now, there's no way to retrieve these separately (core and theme), once we have that in the endpoint we should add these.